### ISSUE TITLE
Modify agent/server retry logic

### ISSUE THREAD

**Issue by 0xdade** [natlas/natlas#443]
**Describe the feature you'd like**
The server and agent communication protocol currently signals whether or not to retry based entirely on a boolean flag in the json response instead of based on an http status code or an http header. This flag may or may not be present, which leads to a lot of conditionals in the agent networking code to account for it. By making use of alternate http status codes and the retry-after header, we can reduce the complexity of the retry logic by handling it uniformly across the application.

**Have you considered alternative ways to get this feature**
Basically we ought to do at least one of these. The retry-after may not be as relevant with the exponential backoff being implemented in the agent, but could give us additional capabilities such as if we know the backend is going to be unreachable for the next 10 minutes or something. Just a thought.
